arrays/find_max_min.cpp

```diff
@@ -0,0 +1,84 @@
+/**
+ * @file
+ * @brief Program to find maximum and minimum element in an array
+ *
+ * @details
+ * This program demonstrates how to find the maximum and minimum values
+ * in an array of elements. It is implemented as a template function so
+ * it works with any comparable data type.
+ *
+ * The algorithm simply iterates through the array once, updating the
+ * maximum and minimum values as needed. This results in \f$O(n)\f$ time
+ * complexity and \f$O(1)\f$ space complexity.
+ */
+
+#include <cassert>   /// for assert
+#include <iostream>  /// for std::cout, std::endl
+#include <vector>    /// for std::vector
+#include <utility>   /// for std::pair
+
+/**
+ * @namespace arrays
+ * @brief Array-based algorithms
+ */
+namespace arrays {
+/**
+ * @namespace find_max_min
+ * @brief Functions for finding maximum and minimum values
+ */
+namespace find_max_min {
+/**
+ * @brief Find maximum and minimum in an array
+ * @tparam T data type of elements (must support comparisons)
+ * @param arr input array
+ * @return std::pair<T, T> containing {max, min}
+ */
+template <typename T>
+std::pair<T, T> findMaxMin(const std::vector<T>& arr) {
+    assert(!arr.empty());  // array must not be empty
```

Comment:
Why not return an empty pair?

Reply:
Because, imagine a user gives an empty array they get an empty pair back, no max, no min. You could make it testable.